lib/view_component/errors.rb

```diff
@@ -17,8 +17,16 @@ def initialize(klass_name)
   end
 
   class TemplateError < StandardError
-    def initialize(errors)
-      super(errors.join("\n"))
+    def initialize(errors, templates = nil)
+      message = errors.join("\n")
+
+      if templates
+        message << "\n"
+        message << "Templates:"
+        message << templates.map(&:inspect).join("\n")
```

Comment:
do we want `inspect` output, or just the template name?

Reply:
All of the inspect output, as it shows what type of templates there are for the component and where they are defined:

```
Template file and inline render method found for variant 'phone' in VariantTemplateAndInlineVariantTemplateComponent. There can only be a template file or inline render method per variant.

Templates:

#<ViewComponent::Template:0x000000012a9980d8 @component=VariantTemplateAndInlineVariantTemplateComponent, @type=:file, @this_format=:html, @variant=:phone, @lineno=0, @path=\"/Users/joelhawksley/github/view_component/test/sandbox/app/components/variant_template_and_inline_variant_template_component.html+phone.erb\", @extension=\"erb\", @source=nil, @method_name=nil, @defined_on_self=true, @source_originally_nil=true, @call_method_name=\"call_phone\">

#<ViewComponent::Template:0x000000012a997ef8 @component=VariantTemplateAndInlineVariantTemplateComponent, @type=:inline_call, @this_format=:html, @variant=:phone, @lineno=nil, @path=nil, @extension=nil, @source=nil, @method_name=:call_phone, @defined_on_self=true, @source_originally_nil=true, @call_method_name=:call_phone>
```